test/package-test.js

```diff
@@ -0,0 +1,172 @@
+
+const Hapi = require('hapi');
+const lab = exports.lab = require('lab').script();
+const code = exports.code = require('code');
+const hapiSlack = require('../');
+let server;
+
+lab.afterEach((done) => {
+  server.stop(() => {
+    done();
+  });
+});
+lab.test('converts a basic message passed as string ', (done) => {
+  server = new Hapi.Server({});
+  server.connection({ port: 8080 });
+  server.register({
+    register: hapiSlack,
+    options: {}
+  }, () => {
+    const packetString = server.makeSlackPayload(['test'], 'a string');
+    code.expect(typeof packetString).to.equal('string');
+    const packet = JSON.parse(packetString);
+    code.expect(packet.attachments.length).to.equal(1);
+    code.expect(packet.attachments[0].fields.length).to.equal(1);
+    code.expect(packet.attachments[0].title).to.equal('a string');
+    code.expect(packet.attachments[0].fields[0].title).to.equal('Tags');
+    done();
+  });
+});
+lab.test('lets you post an object as the message', (done) => {
+  server = new Hapi.Server({});
+  server.connection({ port: 8080 });
+  server.register({
+    register: hapiSlack,
+    options: {}
+  }, () => {
+    const packet = JSON.parse(server.makeSlackPayload([], {
+      data: 'this is an object'
+    }));
+    code.expect(packet.attachments[0].mrkdwn_in[0]).to.equal('text');
+    code.expect(packet.attachments[0].text).to.equal('``` {\n  "data": "this is an object"\n} ```');
+    done();
+  });
+});
+lab.test('"error" tag will set the "danger" color option', (done) => {
+  server = new Hapi.Server({});
+  server.connection({ port: 8080 });
+  server.register({
+    register: hapiSlack,
+    options: {}
+  }, () => {
+    const packet = JSON.parse(server.makeSlackPayload(['error'], 'some text'));
+    code.expect(packet.attachments[0].color).to.equal('danger');
+    done();
+  });
+});
+lab.test('warning tags will have a yellow swatch', (done) => {
+  server = new Hapi.Server({});
+  server.connection({ port: 8080 });
+  server.register({
+    register: hapiSlack,
+    options: {}
+  }, () => {
+    const packet = JSON.parse(server.makeSlackPayload(['warning'], 'test msg'));
+    code.expect(packet.attachments[0].color).to.equal('warning');
+    done();
+  });
+});
+lab.test('"success" tags will have a "good" color', (done) => {
+  server = new Hapi.Server({});
+  server.connection({ port: 8080 });
+  server.register({
+    register: hapiSlack,
+    options: {}
+  }, () => {
+    const packet = JSON.parse(server.makeSlackPayload(['success'], 'a string'));
+    code.expect(packet.attachments[0].color).to.equal('good');
+    done();
+  });
+});
+lab.test('lets you post an object with a special "message" field', (done) => {
+  server = new Hapi.Server({});
+  server.connection({ port: 8080 });
+  server.register({
+    register: hapiSlack,
+    options: {}
+  }, () => {
+    const packet = JSON.parse(server.makeSlackPayload([], {
+      message: 'this is the message that was pulled out of the object below',
```

Comment:
need a test to test adding the fallback field (will get covered if you do a deep equal on the object)